### PR TITLE
ast: dump TypeAlias

### PR DESCRIPTION
**Pull request by jdupak**
Implements ast dump for TypeAlias

### CODE REVIEW

File: gcc/rust/ast/rust-ast-dump.cc [hunk 1]
```diff
@@ -746,7 +745,24 @@ Dump::visit (Function &function)
 
 void
 Dump::visit (TypeAlias &type_alias)
-{}
+{
+  // TODO does this represent associated type
```
Comment (philberty) [inline gcc/rust/ast/rust-ast-dump.cc:749]: Good catch, luckily with our AST class hierarchy for associated types we have: AST::TraitItemType

File: gcc/rust/ast/rust-ast-dump.cc [hunk 2]
```diff
@@ -746,7 +745,24 @@ Dump::visit (Function &function)
...
+  // Visibility? type IDENTIFIER GenericParams? ( : TypeParamBounds )?
+  // WhereClause? ( = Type)? ;
+
+  if (type_alias.has_visibility ())
+    emit_visibility (type_alias.get_visibility ());
+  stream << "type " << type_alias.get_new_type_name ();
+  if (type_alias.has_generics ())
+    emit_generic_params (type_alias.get_generic_params ());
+  if (type_alias.has_where_clause ())
+    {} // FIXME: WhereClause
```
Comment (philberty) [inline gcc/rust/ast/rust-ast-dump.cc:761]: This could be your next dump it will be tricky but a good learning experience for you.

**Review by philberty — APPROVED** [review]
LGTM

**Comment by philberty** [comment]
This is my fault but we might consider updating our clang-format CI here to a later clang-format I was on an old one I think clang-format 10 and there is a breaking change between 10 and >10 so you might need to fix your formatting here manually for now until we fix this.

File: gcc/rust/ast/rust-ast-dump.cc (hunk 1 above)
Comment (jdupak) [reply gcc/rust/ast/rust-ast-dump.cc:749]: I just follow the syntax description in the Rust reference. It seems quite useful to get all the cases.

File: gcc/rust/ast/rust-ast-dump.cc [hunk 3]
```diff
@@ -746,7 +745,24 @@ Dump::visit (Function &function)
...
+  stream << " = ";
+  type_alias.get_type_aliased ()->accept_vis (*this);
+  stream << ";\n\n";
```
Comment (CohenArthur) [inline gcc/rust/ast/rust-ast-dump.cc:764]: Very tiny nitpick: Why two newlines? Is there a specific reason?

No need to fix this right now, I think it's completely fine to leave it that way. It produces valid Rust code and that's what matters :)

**Review by CohenArthur — APPROVED** [review]
LGTM! Thank you! I'll merge this as soon as the formatting issues are resolved

File: gcc/rust/ast/rust-ast-dump.cc (hunk 3 above)
Comment (jdupak) [reply gcc/rust/ast/rust-ast-dump.cc]: I had an example for that but it actually does not make sense in the general case. Fixed.

**Pull request merged**